pkg/apis/machineconfiguration.openshift.io/v1/helpers.go

```diff
@@ -196,3 +202,40 @@ func IsControllerConfigCompleted(ccName string, ccGetter func(string) (*Controll
 	}
 	return fmt.Errorf("ControllerConfig has not completed: completed(%v) running(%v) failing(%v)", completed, running, failing)
 }
+
+// ProxyInspectTest inspects the proxyTestImage to test if the http proxy is valid.
+func ProxyInspectTest(proxy *configv1.ProxyStatus) error {
+	if proxy == nil {
+		return nil
+	}
+	if err := skopeoInspectUseProxy(proxyTestImage, proxy.HTTPProxy, proxy.HTTPSProxy); err != nil {
+		return err
+	}
+	return nil
+}
+
+func skopeoInspectUseProxy(imageURL, httpProxy, httpsProxy string) error {
+	// proxy error from docker registry starts with proxyErr
+	proxyErr := "proxyconnect"
+	skopeo, err := exec.LookPath("skopeo")
+	if err != nil {
+		return err
+	}
+	imageSpec := fmt.Sprintf("docker://%s", imageURL)
+	var envs []string
+	if httpProxy != "" {
+		envs = append(envs, fmt.Sprintf("HTTP_PROXY=%s", httpProxy))
+	}
+	if httpsProxy != "" {
+		envs = append(envs, fmt.Sprintf("HTTPS_PROXY=%s", httpsProxy))
+	}
+
+	args := []string{"inspect", imageSpec}
+	cmd := exec.Command(skopeo, args...)
+	cmd.Env = envs
+	rawOut, err := cmd.CombinedOutput()
+	if err != nil && strings.Contains(string(rawOut), proxyErr) {
```

Comment:
this if statement should be split up because err might be thrown for a different reason.